### PR TITLE
Sort select/radio fields by translation

### PR DESCRIPTION
**Pull request by jbostoen**
Makes it more natural, sorts select/radio attributes by their translation.
Should we make pull requests to 'master' or 'develop'?

For instance, in English you could have this order:

- arts
- sports
- youth

while in Dutch it would be 

- jeugd (youth)
- kunst (arts)
- sport (sports)

### CODE REVIEW

**Comment by Molkobain** [comment]
_TLDR; Too soon to tell you. I'm curious about @dflaven and @v-dumas thoughts on this._

It would definitively be a property of the AttributeEnum class, but as it would also need its counterpart in the XML definition, which requires to be upgraded to a new version number (current version is 1.5, 1.6 coming with iTop 2.6) for both the compiler & the ITSM Designer to handle it.

As this would be introduced as a new feature, we can imagine that we would also like to be able to choose _on what_ it is sorted (value vs label) but also _how_ (ascending vs descending order).

IMHO, this needs to be discussed.

@dflaven & @v-dumas might have some insights from existing requests on this matter.

**Comment by jbostoen** [comment]
@v-dumas , with backward compatibility in mind, is 'rank' really needed? Couldn't it be assumed that the rank is simple the one you have in the XML file? (unless you're thinking about extensions adding values, but then again, you might need to redefine the entire ranking anyway in the extension)

**Comment by Hipska** [comment]
Why not order on value id then? It is currently only used to define and change/remove the values..

**Comment by v-dumas** [comment]
Good idea if we where starting from scratch, but not very flexible on existing installed base. I doubt that we can change those id within the Standard Datamodel, without breaking a lot of existing customization/extensions.

**Comment by Hipska** [comment]
The same counts if you go from
```xml
<value id="name">name</value>
```
to
```xml
<value id="name">
    <code>name</code>
    <rank>3</rank>
</value>
```

**Comment by jbostoen** [comment]
For compatibility/minimal work, I'd suggest just using one tag (sort_on) with:
- **value** (or 'code' or ...)
- **label**  (or 'translation' or ...)
- **xml** 

By default, it could assume 'value', which is the way it works today. Fully compatible with previous work, I guess.

Or is there an issue I'm missing with this approach?

**Comment by v-dumas** [comment]
Using the XML order instead of a rank tag, is again not a good idea, as it forces full reordering by an extension, which may not know what values you have, so its bad. A rank is not perfect, but still better as most extensions will assume Standard Datamodel + maybe other values brought by other extensions, and will decide to rank their value between 2 standard values assuming they have not been moved. We use ranking with decimal, so you don't need to reorder to insert a value.

**Comment by v-dumas** [comment]
@Hispka, iTop compiler is already able to cope with such double syntax

**Comment by dflaven** [comment]
My thoughs on the topics are simple: a PR is not the place to discuss such internal iTop design topics.
